lib/mri/uri/generic.rb

```diff
@@ -591,7 +591,7 @@ def check_host(v)
       if @opaque
         raise InvalidURIError,
           "can not set host with registry or opaque"
-      elsif parser.regexp[:HOST] !~ v
+      elsif !parser.regexp[:HOST].match v
```

Comment:
If intended. please add a justification with `# Truffle: ...`.
This will also be overwritten on the next MRI stdlib update.